src/AutoMapper/MapperConfiguration.cs

```diff
@@ -108,6 +110,16 @@ public TypeMap ResolveTypeMap(Type sourceType, Type destinationType)
         public TypeMap ResolveTypeMap(TypePair typePair)
         {
             var typeMap = _typeMapPlanCache.GetOrAdd(typePair, _getTypeMap);
+
+            _typeMapPlanCache.AddOrUpdate(typePair, typeMap, (tp, tm) =>
+            {
+                if (tm?.Sealed == false)
+                    tm.Seal(_typeMapRegistry, this);
+                else if (typeMap?.Sealed != true)
+                    Thread.Sleep(1);

```

Comment:
This is needed to pass DynamicMapThreadingIssues.Should_Not_Fail test.
Not best solution to fix multi-threaded problem, but works.

Unless someone tries to go farther then the test does I don't see it as a big issue.